### PR TITLE
ORC-870: Unpin and upgrade `jmh` to 1.37

### PR DESCRIPTION
**Pull request by cxzl25**
### What changes were proposed in this pull request?

This PR aims to unpin and upgrade `jmh` to 1.37.

### Why are the changes needed?

To use the latest bug fixed version.

### How was this patch tested?
```
cd java
./mvnw clean package -Pbenchmark -DskipTests
```

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by mystic-lama** [comment]
@cxzl25 Can we please update the PR description with details on What and Why for the changes ?

File: java/bench/hive/src/java/org/apache/orc/bench/hive/rowfilter/BooleanRowFilterBenchmark.java [hunk 1]
```diff
@@ -91,10 +91,14 @@ public void readOrcNoFilter(Blackhole blackhole, InputState state) throws Except
    * Run this test:
    *  java -cp hive/target/orc-benchmarks-hive-*-uber.jar org.apache.orc.bench.hive.rowfilter.BooleanRowFilterBenchmark
    */
-  public static void main(String[] args) throws RunnerException {
-    new Runner(new OptionsBuilder()
-        .include(BooleanRowFilterBenchmark.class.getSimpleName())
-        .forks(1)
-        .build()).run();
+  public static void main(String[] args) {
```
Comment (dongjoon-hyun) [inline java/bench/hive/src/java/org/apache/orc/bench/hive/rowfilter/BooleanRowFilterBenchmark.java:94]: Do you know why this change is required? Apache ORC code follows the official JMHSample file.
- https://github.com/openjdk/jmh/blob/cb3c3a90137dad781a2a37fda72dc11ebf253593/jmh-samples/src/main/java/org/openjdk/jmh/samples/JMHSample_01_HelloWorld.java#L95

```java
    public static void main(String[] args) throws RunnerException {
        Options opt = new OptionsBuilder()
                .include(JMHSample_01_HelloWorld.class.getSimpleName())
                .forks(1)
                .build();

        new Runner(opt).run();
    }
```

**Review by dongjoon-hyun — COMMENTED** [review]
To @cxzl25 , I updated the PR description for you. BTW, I have a question on this change, https://github.com/apache/orc/pull/1740/files#r1447688404.

File: java/bench/hive/src/java/org/apache/orc/bench/hive/rowfilter/BooleanRowFilterBenchmark.java (hunk 1 above)
Comment (cxzl25) [reply java/bench/hive/src/java/org/apache/orc/bench/hive/rowfilter/BooleanRowFilterBenchmark.java:94]: Because these Benchmarks inherit `extends org.openjdk.jmh.Main`, `org.openjdk.jmh.Main` removed the method throwing `RunnerException` in `1.23`.

https://github.com/openjdk/jmh/commit/4264de9486c32b48da8161e3ac076a0187b4176f#diff-30878bccaff9f11a64415de78f1feebe304334a314d8bb6f8c7bc50bd21aee19L39

**Comment by cxzl25** [comment]
> I updated the PR description for you

Thanks your help. @dongjoon-hyun

File: .github/dependabot.yml [hunk 2]
```diff
@@ -38,6 +35,6 @@ updates:
       # Pin mockito to 4.x
       - dependency-name: "org.mockito"
         versions: "[5.0.0,)"
-      # Pin byte-buddy to 1.12.x aline with mockito
+      # Pin byte-buddy to 1.12.x align with mockito
```
Comment (mystic-lama) [inline .github/dependabot.yml:38]: This is minor but unrelated change and perhaps can be avoided in this PR.
Let's see what other's think.

File: java/bench/hive/src/java/org/apache/orc/bench/hive/rowfilter/BooleanRowFilterBenchmark.java (hunk 1 above)
Comment (dongjoon-hyun) [reply java/bench/hive/src/java/org/apache/orc/bench/hive/rowfilter/BooleanRowFilterBenchmark.java:94]: Does it mean the official `JMHSample_01_HellowWorld.java` is also broken?

Comment (cxzl25) [reply java/bench/hive/src/java/org/apache/orc/bench/hive/rowfilter/BooleanRowFilterBenchmark.java:94]: `JMHSample_01_HelloWorld` does not inherit jmh.Main ( `extends org.openjdk.jmh.Main` )

File: .github/dependabot.yml (hunk 2 above)
Comment (dongjoon-hyun) [reply .github/dependabot.yml:38]: +1 for @mystic-lama 's comment. Please recover this line.

File: java/bench/hive/src/java/org/apache/orc/bench/hive/rowfilter/BooleanRowFilterBenchmark.java (hunk 1 above)
Comment (dongjoon-hyun) [reply java/bench/hive/src/java/org/apache/orc/bench/hive/rowfilter/BooleanRowFilterBenchmark.java:94]: Oh, got it. Thanks. Now, I understand.

File: .github/dependabot.yml (hunk 2 above)
Comment (dongjoon-hyun) [reply .github/dependabot.yml]: I removed it, @cxzl25 .

Comment (cxzl25) [reply .github/dependabot.yml]: Thanks. @dongjoon-hyun

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. 

Let me do the final review and merge this.

**Comment by dongjoon-hyun** [comment]
Merged to main/2.0 for Apache ORC 2.0.0.

Thank you, @cxzl25 and @mystic-lama .
